### PR TITLE
BUGFIX: Set the subpackage key when setting the package key in forward()

### PR DESCRIPTION
**Pull request by cerlestes**
There's no logical case where you'd want to forward to a different package but stay within the same subpackage key. That's why the subpackage key should be (re)set whenever the package is changed when forwarding.

### CODE REVIEW

File: TYPO3.Flow/Classes/TYPO3/Flow/Mvc/Controller/AbstractController.php [hunk 1]
```diff
@@ -196,8 +196,6 @@ protected function forward($actionName, $controllerName = null, $packageKey = nu
         }
         if ($packageKey !== null) {
             $nextRequest->setControllerPackageKey($packageKey);
-        }
```
Comment (kitsunet) [inline TYPO3.Flow/Classes/TYPO3/Flow/Mvc/Controller/AbstractController.php]: This basically solves the case I guess but opens another bug if you don't set $packageKey but $subpackagekey because that won't be picked up then

Comment (cerlestes) [inline TYPO3.Flow/Classes/TYPO3/Flow/Mvc/Controller/AbstractController.php]: You cannot set the subpackageKey without setting the packageKey though.

Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Mvc/Controller/AbstractController.php]: Right, subpackageKey is not a method argument, but is directly derived from packageKey.

Test needs to be adjusted though, and there seems to be an unrelated CGL error - should maybe be rebased

Also, does this maybe make sense for 2.3/3.0 too?

**Comment by albe** [comment]
Closing in favor of #319
